### PR TITLE
github CI action

### PR DESCRIPTION
**Pull request by chcg**
- added appveyor.yml with preparation for automatic github deployment on tagging (needs secure token at the TODO, see also https://www.appveyor.com/docs/deployment/github/#provider-settings how to generate Personal API access token at https://github.com/settings/tokens and encrypt it)

- avoid vs2015 compiler warnings about hidden global vars
- enabled multithreaded builds in vcxproj
- enabled sdl checks in vcxproj
- corrected platform toolset for x64 release build
- corrected copy action of readme.txt

@donho see https://ci.appveyor.com/project/chcg/selectnlaunch/builds/35802760

### CODE REVIEW

**Comment by chcg** [comment]
@donho Added arm64, see https://ci.appveyor.com/project/chcg/selectnlaunch/builds/38972754

**Comment by chcg** [comment]
@donho Replaced appveyor by GH action and updated PR to VS2022

File: src/SelectNLaunch.cpp [hunk 1]
```diff
@@ -216,25 +216,25 @@ HWND getCurrentScintillaHandle() {
 
 
 
-int getCmdsFromConf(const TCHAR *confPath, CmdParam *cmdParam, int /*maxNbCmd*/)
+int getCmdsFromConf(const TCHAR *confPathValue, CmdParam * /*cmdParamValue*/, int /*maxNbCmd*/)
```
Comment (donho) [inline src/SelectNLaunch.cpp:219]: `cmdParam` is used in line [#233 & #234](https://github.com/npp-plugins/selectnlaunch/pull/1/files#diff-b4bd6acba2ef7d5b47704025f66666dba05e1ad432e55ddfb0181d007dace1dcR233) so it cannot be removed.

Comment (donho) [reply src/SelectNLaunch.cpp:219]: And why we should change name `confPath` to `confPathValue`?

**Review by donho — CHANGES_REQUESTED** [review]
Restore SelectNLaunch.h & SelectNLaunch.cpp please.

File: src/SelectNLaunch.cpp (hunk 1 above)
Comment (chcg) [reply src/SelectNLaunch.cpp:219]: Global declarations are hidden (https://github.com/npp-plugins/selectnlaunch/blob/master/src/SelectNLaunch.cpp#L33 for confPath and https://github.com/npp-plugins/selectnlaunch/blob/master/src/SelectNLaunch.cpp#L64 for cmdParam) , see unfortunately in german:


```
1>C:\Users\...\nppplugin_selectandlaunch\selectnlaunch\src\SelectNLaunch.cpp(219,34): error C2220: Die folgende Warnung wird als Fehler behandelt.
1>C:\Users\...\nppplugin_selectandlaunch\selectnlaunch\src\SelectNLaunch.cpp(219,34): warning C4459: Deklaration von "confPath" blendet globale Deklaration aus
1>C:\Users\...\nppplugin_selectandlaunch\selectnlaunch\src\SelectNLaunch.cpp(33,16): message : Siehe Deklaration von "confPath"
1>C:\Users\...\nppplugin_selectandlaunch\selectnlaunch\src\SelectNLaunch.cpp(219,54): warning C4459: Deklaration von "cmdParam" blendet globale Deklaration aus
1>C:\Users\...\nppplugin_selectandlaunch\selectnlaunch\src\SelectNLaunch.cpp(64,10): message : Siehe Deklaration von "cmdParam"

```

Comment (donho) [reply src/SelectNLaunch.cpp:219]: @chcg 
> Global declarations are hidden for confPath and for cmdParam.

You're right about it.

**Review by donho — APPROVED** [review]

**Comment by donho** [comment]
@chcg 
However, we have still the following error on CI:
```
The build phase is set to "MSBuild" mode (default), but no Visual Studio project or solution files were found in the root directory. If you are not building Visual Studio project switch build mode to "Script" and provide your custom build command.
```

Could you check it please?

**Comment by donho** [comment]
BTW, why don't we follow the other schema of projects - Add Github CI and remove Appveyor ?

**Comment by chcg** [comment]
@donho appveyor.yml is replaced by github action, but the project is still registered at appveyor and therefore the failure from appveyor. You have to delete it at https://ci.appveyor.com/project/donho/selectnlaunch/settings/delete.

Updated N++ header files because Sci_Position.h was missing by your last update.

**Comment by donho** [comment]
> @donho appveyor.yml is replaced by github action, but the project is still registered at appveyor and therefore the failure from appveyor. You have to delete it at https://ci.appveyor.com/project/donho/selectnlaunch/settings/delete.

It's been deleted.

> Updated N++ header files because Sci_Position.h was missing by your last update.

Could you add both Sci_Position.h & Scintilla.h from https://github.com/notepad-plus-plus/notepad-plus-plus/tree/master/scintilla/include please?

**Comment by chcg** [comment]
@donho Update of all N++ header files from trunk already done.

**Comment by donho** [comment]
Thank you @chcg 
But **All checks have failed** still bothering. Can we relaunch the Github CI for validating this PR?

**Comment by donho** [comment]
> But All checks have failed still bothering. Can we relaunch the Github CI for validating this PR?

Sorry to bother you with that.
It's because x86 release cannot be compiled by my VS2022 with the following message: 
```
Severity	Code	Description	Project	File	Line	Suppression State
Error	MSB8020	The build tools for Visual Studio 2013 - Windows XP (Platform Toolset = 'v120_xp') cannot be found. To build using the v120_xp build tools, please install Visual Studio 2013 - Windows XP build tools.  Alternatively, you may upgrade to the current Visual Studio tools by selecting the Project menu or right-click the solution, and then selecting "Retarget solution".	SelectNLaunch	C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Microsoft\VC\v170\Microsoft.CppBuild.targets	460	
```

Also, the ARM 64 release is not present for being built:

image

**Comment by chcg** [comment]
You can see it running in my repo: https://github.com/chcg/selectnlaunch/actions. The last one https://github.com/chcg/selectnlaunch/actions/runs/6511248970. Seems you missed somehow the update of the vcxproj file which had the platformtools v120_xp before but is updated to v143 (VS2022) by this PR.

I think you need to approve it that this PR is allowed to run the action (https://docs.github.com/en/actions/managing-workflow-runs/approving-workflow-runs-from-public-forks) it should then be possible also to rerun it from the action tab https://docs.github.com/en/actions/managing-workflow-runs/re-running-workflows-and-jobs
Maybe you have to enable actions for this repo, see https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/enabling-features-for-your-repository/managing-github-actions-settings-for-a-repository.

**Comment by donho** [comment]
Thank you @chcg 
Just check the PR on PC of release, it compiles like a charm.

**Pull request closed without merging**